underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -149,29 +150,48 @@ public static S3AUnderFileSystem createInstance(AlluxioURI uri) {
     }
 
     AmazonS3Client amazonS3Client = new AmazonS3Client(credentials, clientConf);
+    // Set a custom endpoint.
     if (Configuration.containsKey(PropertyKey.UNDERFS_S3_ENDPOINT)) {
       amazonS3Client.setEndpoint(Configuration.get(PropertyKey.UNDERFS_S3_ENDPOINT));
     }
+    if (Configuration.containsKey(PropertyKey.UNDERFS_S3_DISABLE_DNS_BUCKETS)) {
```

Comment:
Since you have already set a default value for the key, its unnecessary to check the existence.